controller/dataplane/controller_reconciler_utils.go

```diff
@@ -77,6 +84,175 @@ func (r *Reconciler) ensureDataPlaneAddressesStatus(
 	return false, nil
 }
 
+// ensureMappedConfigMapToKongPluginInstallationForDataPlane ensures that the KongPluginInstallation
+// resources referenced by the DataPlane are resolved and DataPlane is configured to use them.
+// During resolving for each DataPlane based on each instance of KongPluginInstallation
+// ConfigMap is created and mounted. The DataPlane manages its lifecycle. It returns a slice
+// of custom plugins that are intended to be used to generate a Deployment.
+func ensureMappedConfigMapToKongPluginInstallationForDataPlane(
+	ctx context.Context, logger logr.Logger, c client.Client, dataplane *operatorv1beta1.DataPlane,
+) (cps []customPlugin, requeue bool, err error) {
+	configMapsOwned, err := findCustomPluginConfigMapsOwnedByDataPlane(ctx, c, dataplane)
+	if err != nil {
+		return nil, false, err
+	}
+	configMapsToRetain := make(map[types.NamespacedName]struct{}, len(configMapsOwned))
+
+	for _, kpiNN := range dataplane.Spec.PluginsToInstall {
+		kpiNN := types.NamespacedName(kpiNN)
+		if kpiNN.Namespace == "" {
+			kpiNN.Namespace = dataplane.Namespace
+		}
+
+		cp, requeue, err := populateDedicatedConfigMapForKongPluginInstallation(
```

Comment:
This is OK but I'd suggest to be cautious about redefining the named return variables.

Reply:
Improved, tbh I don't like naked returns, but named returns for me are quite useful as a documentation 😉